### PR TITLE
Mutations/zach

### LINKED ISSUE #32: can create a resource

### PR DESCRIPTION
**Pull request by zachjamesgreen**
## Checklist
- [x] Tests written where required
- [x] SimpleCov at 81.7%
- [x] Squash & Rebase
- [x] All tests passing
- [ ] Prettier has been run
- [x] Reek has been run
- [x] Circle CI builds are passing

## Changes
Can Create, Edit, and Delete a Project resource
Can Create, Delete a Project Repo
Return error if user can not be found

closes #32 #33 #34

### CODE REVIEW

File: app/controllers/graphql_controller.rb [hunk 1]
```diff
@@ -41,6 +48,15 @@ def prepare_variables(variables_param)
     end
   end
 
+  def check_header
+    x = false
```
Comment (twarbelow) [inline app/controllers/graphql_controller.rb]: could we change this variable name to something more informative?

File: app/graphql/mutations/create_collaborator.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+module Mutations
+  class CreateCollaborator < BaseMutation
+    argument :user_id, ID, required: true
+    argument :collab_id, ID, required: true
+    argument :project_id, ID, required: true
+
+    field :errors, [String], null: false
+    field :collaborator, Types::UserType, null: false
+
+    def resolve(user_id:, collab_id:, project_id:)
```
Comment (twarbelow) [inline app/graphql/mutations/create_collaborator.rb]: is this supposed to do something...?

File: app/graphql/mutations/create_project_repo.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    argument :user_id, ID, required: true
+    argument :project_id, ID, required: true
+    argument :repo_name, ID, required: true
+
+    # field :errors, [String], null: false
+    # user must be collaborator on project
+    # return the repo if it was created
+    # return the errors if the repo was not created
+
+    def resolve(info)
```
Comment (twarbelow) [inline app/graphql/mutations/create_project_repo.rb]: is this supposed to do something?

File: app/graphql/mutations/create_resource.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        content: content
+      )
+      resource.tags = tags if tags.present?
+      
+      if resource.save
+        return_info(resource)
+      else
+        return_info(nil, errors: resource.errors.full_messages)
+      end
+    rescue ArgumentError => e
```
Comment (twarbelow) [inline app/graphql/mutations/create_resource.rb]: is the e supposed to stand for errors?

File: spec/request/queries/user/get_user_spec.rb [hunk 5]
```diff
@@ -23,4 +23,7 @@ def query
     expect(response.body).to include(@user.id.to_s)
     expect(response.body).to include(@user.github_handle)
   end
+
+  it 'should return error if user is not found'
```
Comment (twarbelow) [inline spec/request/queries/user/get_user_spec.rb]: want to put a test here?

**Review by twarbelow — CHANGES_REQUESTED** [review]
please respond to comments :)

File: app/graphql/mutations/create_collaborator.rb (hunk 2 above)
Comment (zachjamesgreen) [reply app/graphql/mutations/create_collaborator.rb]: Its from when i got started on collaborator mutations

File: app/controllers/graphql_controller.rb (hunk 1 above)
Comment (zachjamesgreen) [reply app/controllers/graphql_controller.rb]: Changing this. Probably not going to use it anyway

**Review by twarbelow — APPROVED** [review]
thanks for the changes

**Pull request merged**